recommendations_Infra_demo/recommendations_demo.sh

```diff
@@ -0,0 +1,430 @@
+}
+
+# By default we start the demo & experiment and we dont expose prometheus port
+prometheus=0
+cluster_monitoring_setup=1
+demo_monitoring_setup=1
+start_demo=1
+validateRecommendations=0
+CLUSTER_NAME=""
+#echo "clustername is $CLUSTER_NAME"
+#echo "setup is $demo_monitoring_setup"
+# Iterate through the commandline options
+while getopts o:c:d:prstaug-: gopts
+do
+
+	 case ${gopts} in
+         -)
+                case "${OPTARG}" in
+                        visualize)
+                                visualize=1
+				;;
+			dataDrivenRecommendations)
+				dataDrivenRecommendations=1
+				;;
+			monitorRecommendations)
+				monitorRecommendations=1
+                                ;;
+			compareRecommendations)
+				compareRecommendations=1
+				;;
+			demoBenchmark)
+				demoBenchmark=1
+				;;
+			k8ObjectName=*)
+				k8ObjectName=${OPTARG#*=}
+				;;
+			k8ObjectType=*)
+				k8ObjectType=${OPTARG#*=}
+				;;
+			mode=*)
+				mode=${OPTARG#*=}
+				;;
+			dataDir=*)
+				resultsDir=${OPTARG#*=}
+				;;
+			summarizeClusters)
+				summarizeClusters=1
+				cluster_monitoring_setup=0
+				demo_monitoring_setup=0
+				;;
+			summarizeNamespaces)
+                                summarizeNamespaces=1
+                                cluster_monitoring_setup=0
+                                demo_monitoring_setup=0
+                                ;;
+			summarizeAll)
+				summarizeAll=1
+				cluster_monitoring_setup=0
+                                demo_monitoring_setup=0
+                                ;;
+
+			clusterName=*)
+				echo "checking.."
+				CLUSTER_NAME=${OPTARG#*=}
+				echo "clusterName is ${OPTARG#*=}"
+				;;
+			namespaceName=*)
+				NAMESPACE_NAME=${OPTARG#*=}
+				;;
+			validate)
+				validateRecommendations=1
+				;;
+
+                        *)
+				;;
+                esac
+                ;;
+
+	o)
+		AUTOTUNE_DOCKER_IMAGE="${OPTARG}"
+		;;
+	p)
+		prometheus=1
+		;;
+	r)
+		cluster_monitoring_setup=0
+		demo_monitoring_setup=1
+		;;
+	s)
+		start_demo=1
+		;;
+	t)
+		start_demo=0
+		;;
+	c)
+		CLUSTER_TYPE="${OPTARG}"
+		;;
+	d)
+		DURATION="${OPTARG}"
+		;;
+	a)
+		cluster_monitoring_setup=0
+		demo_monitoring_setup=0
+		;;
+	u)
+		setRecommendations=1
+		cluster_monitoring_setup=0
+		demo_monitoring_setup=0
+		;;
+	g)
+		getMetricsRecommendations=1
+		cluster_monitoring_setup=0
+		demo_monitoring_setup=0
+		;;
+	*)
+		usage
+		;;
+	esac
+done
+
+#Todo
+# Options
+# Generate recommendations for the data given.
+
+# Monitor the metrics in a cluster to generate recommendations
+
+# Benchmark specific recommendations
+
+# Copy the previous recommendationsOutput.csv and experimentOutput.csv into another for future purpose.
+if [ -e "recommendationsOutput.csv" ]; then
+	mv recommendationsOutput.csv recommendationsOutput-$(date +%Y%m%d).csv
+fi
+if [ -e "experimentOutput.csv" ]; then
+        mv experimentOutput.csv experimentOutput-$(date +%Y%m%d).csv
+fi
+
+if [ ${start_demo} -eq 1 ]; then
+	monitoring_demo_start
```

Comment:
Script should exit with 0 (success) or 1 (failure), this is required to automate this on jenkins for --validate option

Reply:
@kusumachalasani - Has this been done? I will need this to include the actual recommendation validation (--validate) in the jenkins job. How long does this take to run?

Reply:
Yes, exit codes are incorpoated. It takes around 30mins to complete the run to validate 4 csv files (which has 15 days data) as the recommendations are generated for every timestamp.